File: Hammerspoon/MJAutoLaunch.m
#import "MJAutoLaunch.h"
#import <ServiceManagement/ServiceManagement.h>

BOOL MJAutoLaunchGet(void) {
    SMAppService *service = [SMAppService mainAppService];
    return service.status == SMAppServiceStatusEnabled;
}

void MJAutoLaunchSet(BOOL opensAtLogin) {
    SMAppService *service = [SMAppService mainAppService];
    NSError *error = nil;
    if (opensAtLogin) {
        [service registerAndReturnError:&error];
    } else {
        [service unregisterAndReturnError:&error];
    }
    if (error) {
        NSLog(@"MJAutoLaunch: %@", error);
    }
}

File: Hammerspoon/MJAutoLaunch.swift
import Foundation
import ServiceManagement

@_cdecl("MJAutoLaunchGet")
func MJAutoLaunchGet() -> Bool {
    SMAppService.mainApp.status == .enabled
}

@_cdecl("MJAutoLaunchSet")
func MJAutoLaunchSet(_ opensAtLogin: Bool) {
    let service = SMAppService.mainApp
    do {
        if opensAtLogin {
            try service.register()
        } else {
            try service.unregister()
        }
    } catch {
        NSLog("MJAutoLaunch: %@", error as NSError)
    }
}
